### DISCUSSION TITLE
Cannot import "create" as named import.

### DISCUSSION THREAD

**Discussion by wuarmin** [pmndrs/zustand#1628] [General]
## Summary
If I try to import `create` from `zustand` as a named import, as described in the docs, an error is thrown:
```javascript
import { create } from "zustand"
```
```txt
Uncaught SyntaxError: ambiguous indirect export: create bear-store.js:1:9
```

## Workaround
```javascript
import create from "zustand"
```
but this is marked as deprecated:
```javascript
export declare const create: Create;
/**
 * @deprecated Use `import { create } from 'zustand'`
 */
declare const _default: Create;
export default _default;
```

## Reproduction
```javascript
//bear-store.js
import { create } from "zustand";

const useBearStore = create((set) => ({
  bears: 0,
  fish: 0,
  actions: {
    increasePopulation: (by) =>
      set((state) => ({ bears: state.bears + by })),
    eatFish: () => set((state) => ({ fish: state.fish - 1 })),
    removeAllBears: () => set({ bears: 0 }),
  },
}))

export const useBears = () => useBearStore((state) => state.bears)
export const useFish = () => useBearStore((state) => state.fish)
export const useBearActions = () => useBearStore((state) => state.actions)
```

```javascript
import { useBearActions } from "./bear-store" // error is thrown in application
```
## Environment
Vite.js
zustand 4.3.3

Thanks

**Comment by dai-shi** [comment]
Thanks for reporting! This is very interesting. I wonder if there's a solution.
But, what throws `Uncaught SyntaxError: ambiguous indirect export` ?

This is only an issue with vite, right? Can you share a small repo to reproduce the error?

Is anyone familiar with this error?

**Reply by wuarmin** [reply]
I solved the issue. I think it was an issue with two different versions of `zustand` in my pnpm monorepo. I had a package "shared-ui" with `4.1.4_react@18.2.0` installed. The package had a `peerDependency` of `"zustand": "^4.1.4"`. The app, which has the dependency `shared-ui` had `zustand: 4.3.3_react@18.2.0` installed. I think this was the reason.

**Comment by francescoagati** [comment]
same problem

https://stackblitz.com/edit/react-ts-wk6fah?file=App.tsx,package.json,index.tsx

**Reply by dai-shi** [reply]
<img width="401" alt="image" src="https://user-images.githubusercontent.com/490574/224054645-46ed690e-ef7c-4078-8b7d-edc97e3d6d8d.png">

But, we don't have import.meta in https://unpkg.com/browse/zustand@4.3.6/index.js
I suspect stackblitz does transpile ESM code into CJS, but it doesn't deal with import.meta.

And, it's a different problem.

**Reply by francescoagati** [reply]
stackblitz should use vite.js maybe i must use legacy plugin

**Reply by francescoagati** [reply]
but zustand 3 work with stackblitz but no 4

**Reply by dai-shi** [reply]
We didn't use import.meta for ESM in v3. Ironically, that import.meta thing is proposed by vitejs and we follow it.

**Reply by francescoagati** [reply]
yes is strange. see version 3 https://stackblitz.com/edit/react-zustand-immer?file=src%2FApp.js
